shm/src/store.cpp

```diff
@@ -35,4 +43,39 @@ void Store::nextDay() {
     for (auto el : cargo_) {
         *el.get() += 1;
     }
+}
+
+cargoPtr Store::GetCargo(const size_t pos) {
+    if (pos >= cargo_.size())
+        return cargo_.at(cargo_.size() - 1);
+    return cargo_.at(pos);
+}
+
+void Store::GenerateCargo() {
+    // size_t number = GenerateRandom(10);
+    // double basePrice = GenerateRandom(100);
+    // for (auto el : fruitsNames) {
+    // Fruit fr
+    // }
+    // cargo_.emplace_back(&cargo1);
+}
+
+cargoPtr Store::FindMatchCargo(cargoPtr cargo) {
+    for (auto el : cargo_) {
+        if (*el == *cargo) {
+            return el;
+        }
+    }
+    return nullptr;  // czy tak może być? Bo coś tu chyba musimy zwracać, jak się nie znajdzie? Ale czy nullptr to dobra myśl?
```

Comment:
not sure :) findMatchCargo in Ship class has no return statement in case of no match